core/src/main/java/ar/com/wolox/wolmo/core/util/ImageProvider.kt

```diff
@@ -0,0 +1,247 @@
+package ar.com.wolox.wolmo.core.util
+
+import android.content.Context
+import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.net.Uri
+import android.os.Environment
+import android.provider.MediaStore
+import androidx.annotation.IntRange
+import androidx.annotation.StringDef
+import androidx.annotation.StringRes
+import androidx.fragment.app.Fragment
+import ar.com.wolox.wolmo.core.di.scopes.ApplicationScope
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.IOException
+import javax.inject.Inject
+
+/**
+ * Utils class to manipulate images, through {@link Bitmap}s or their corresponding {@link Uri}, and
+ * for retrieving pictures from gallery/taking them from the camera.
+ */
+@ApplicationScope
+class ImageProvider @Inject constructor(val context: Context, val toastFactory: ToastFactory, val wolmoFileProvider: WolmoFileProvider) {
+
+    @Retention(AnnotationRetention.SOURCE)
+    @StringDef(PNG, JPG)
+    annotation class ImageFormat
+
+    /*
+     * Image compression formats supported.
+     */
+
+    companion object {
+        const val PNG = "png"
+        const val JPG = "jpg"
+
+        /**
+         * Prevents quality from being outside 0...100 range.
+         *
+         * @param quality target quality
+         * @return if below 0, returns 0, if above 100, return 100, else returns `quality` param
+         */
+        private fun sanitizeQuality(quality: Int): Int = when {
```

Comment:
No need to specify `Int` as the return type.

Reply:
ok :+1: